Update how_to_add_packages.md (#5962)
conan-io/conan-center-index: docs/how_to_add_packages.md

  * [More Information about Recipes](#more-information-about-recipes)
    * [The recipe folder](#the-recipe-folder)
    * [The version folder/s](#the-version-folders)
    * [The conanfile.py and `test_package` folder](#the-conanfilepy-and-test_package-folder)
    * [The `conandata.yml`](#the-conandatayml)
  * [How to provide a good recipe](#how-to-provide-a-good-recipe)
  * [Test the recipe locally](#test-the-recipe-locally)
    * [Updating conan hooks on your machine](#updating-conan-hooks-on-your-machine)
  * [Debugging failed builds](#debugging-failed-builds)<!-- endToc -->

## Join the Early Access Program

:one: The first step in adding packages to ConanCenter is requesting access to the **Early Access Program**. To enroll in EAP, please write a comment
requesting access in this GitHub [issue](https://github.com/conan-io/conan-center-index/issues/4). The EAP was designed to onboard authors
to the new process.

All EAP requests are reviewed and approved (or denied) every week, thus your request can take one week to be approved, so don't worry. This
process helps conan-center-index against spam and malicious code.

## Submitting a Package

> :two: To contribute a packages, you can submit a [Pull Request](https://github.com/conan-io/conan-center-index/pulls) to this GitHub repository https://github.com/conan-io/conan-center-index.

Reviewer: ```suggestion
:two: To contribute a package, you can submit a [Pull Request](https://github.com/conan-io/conan-center-index/pulls) to this GitHub repository https://github.com/conan-io/conan-center-index.
```

The specific steps to add new packages are:
* Fork the [conan-center-index](https://github.com/conan-io/conan-center-index) git repository, and then clone it locally.

> * Create a new folder with the Conan package recipe (conanfile.py) in the correct folder.

Reviewer: ```suggestion
* Create a new folder with the Conan package recipe (`conanfile.py`) in the correct folder.
```

* Make sure you are using the latest [Conan client](https://conan.io/downloads) version, as recipes might evolve introducing features of the newer Conan releases.
* Commit and Push to GitHub then submit a pull request.
* Our automated build service will build 100+ different configurations, and provide messages that indicate if there were any issues found during the pull request on GitHub.

> :three: When the pull request is [reviewed and merged](review_process.md), those packages are published to [JFrog ConanCenter](https://conan.io/center/).

Reviewer: ```suggestion
:three: When the pull request is [reviewed and merged](review_process.md), those packages are published to [JFrog ConanCenter](https://conan.io/center/) and available for everyone.
```

|           +-- test_package
```

### The version folder(s)

The system supports to use the same recipe for several versions of the library and also to create different recipes for different versions

- **1 version => 1 recipe**

>   When the recipe changes significantly between different library versions and reusing the recipe is not worth it, you can create a folder for each new version and
>   create both inside the `conanfile.py` and the `test_package` in the version folder:

Reviewer: ```suggestion
  When the recipe changes significantly between different library versions and reusing the recipe is not worth it, it is recommended to create different folders and group together the
  versions that can share the same recipe. Each of these folders will require its own `conanfile.py` and `test_package` folder:
```

- **N versions => 1 recipe**

>    Create a folder named `all` (just a convention) and put both the `conanfile.py` and the `test_package` folder there. With this approach, the `conanfile.py`
>    won’t declare the `version` attribute.

Reviewer: ```suggestion
   Create a folder named `all` (just a convention) and put both the `conanfile.py` and the `test_package` folder there.
```

We can skip that part, we never add the `version` attribute, right?

Author: That's an excellent question. I didn't fact check this it was already there

Author: https://github.com/conan-io/conan-center-index/blob/6d1c2e29d7623af9654f614c7a7d1041a5e1d1db/recipes/zlib/1.2.11/conanfile.py#L8

I don't think this pattern is used at all

Then in your `conanfile.py` method, it has to be used to download the sources:

```py
 def source(self):
     tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
```

> > :warning: Please be aware that `conan-center-index` only builds from sources and pre-compiled binaries are not acceptable. For more information see our [packaging policy](https://github.com/conan-io/conan-center-index/blob/master/docs/packaging_policy.md).

Reviewer 2: There are recipes that package binaries in this repo: msys2 and zulu-openjdk.
Perhaps this line should state that cci tries to build from source as much as possible and that packaging binaries is discouraged.

Author: #5792 is a recent example of this too
